src/Sylius/Bundle/ResourceBundle/Behat/DefaultContext.php

```diff
@@ -456,6 +456,37 @@ protected function replaceColumnNamesWithColumnIds(NodeElement $table, array $fi
     }
 
     /**
+     * Callback can return any value, to stop waiting use anonymous function provided as first argument.
+     *
+     * @param callable $callback
+     * @param int $limit
+     * @param int $delay In miliseconds
+     *
+     * @return mixed
+     *
+     * @throws \RuntimeException If timeout was reached
+     */
+    protected function waitFor(callable $callback, $limit = 10, $delay = 100)
+    {
+        $forceBreak = false;
+        $emergencyExit = function () use (&$forceBreak) {
+            $forceBreak = true;
+        };
+
+        for ($i = 0; $i < $limit; ++$i) {
+            $payload = $callback($emergencyExit);
+
+            if ($forceBreak) {
```

Comment:
Yeah, I made it a little bit overcomplicated. Wanted to add ability to wait for `null` values, but it seems like there's no real use case for this :)